parts/xmem_nozzle.go

```diff
@@ -3733,15 +3749,17 @@ func (xmem *XmemNozzle) handleCasPoisoning(wrappedReq *base.WrappedMCRequest, re
 		// Note: We shouldn't hit this case in practice
 		xmem.Logger().Errorf("extras.cas is not set in req %v. len of extras:%v, isSubDocOp: %v", wrappedReq.Req, len(wrappedReq.Req.Extras), isSubDocOp)
 	}
-
+	srcvb := wrappedReq.GetSourceVB()
+	tgtvb := wrappedReq.GetTargetVB()
+	seqno := wrappedReq.Seqno
 	if response.Status == mc.SUCCESS && (sentCas != 0 && sentCas != response.Cas) && !isSubDocOp { //replace mode
 		// Currently CAS regeneration takes place in two scenario's
 		// 1. If SubDoc is used
 		// 2. If there is a CAS poisoned doc and KV's protection mode is set to "replace"
 		// Note that this counter does not account for poisoned CAS's incase of subDocOp
-		return base.ReplaceMode
+		xmem.RaiseEvent(common.NewEvent(common.DocsSentWithPoisonedCas, base.ReplaceMode, xmem, []interface{}{srcvb, tgtvb, seqno}, nil))
 	} else if response.Status == mc.CAS_VALUE_INVALID { //error mode
-		return base.ErrorMode
+		xmem.RaiseEvent(common.NewEvent(common.DocsSentWithPoisonedCas, base.ErrorMode, xmem, []interface{}{srcvb, tgtvb, seqno}, nil))
```

Comment:
Same - should be `return base.ErrorMode`

Reply:
Done